Project.toml

```diff
@@ -14,43 +14,22 @@ SciMLBase = "0bca4576-84f4-4d90-8ffe-ffa030f20462"
 SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
 
 [weakdeps]
-Enzyme = "7da242da-08ed-463a-9acd-ee780be4f1d9"
-FiniteDiff = "6a86dc24-6348-571c-b903-95158fe2bd41"
-ForwardDiff = "f6369f11-7733-5829-9624-2563aa707210"
+DifferentiationInterface = "a0c0ee7d-e4b9-4e03-894e-1c5f64a51d63"
 ModelingToolkit = "961ee093-0014-501f-94e3-6117800e7a78"
-ReverseDiff = "37e2e3b7-166d-5795-8a7a-e32c996b4267"
-SparseDiffTools = "47a9eef4-7e08-11e9-0b38-333d64bd3804"
-Symbolics = "0c5d862f-8b57-4792-8d23-62f2024744c7"
-Tracker = "9f7883ad-71c0-57eb-9f7f-b5c9e6d3789c"
-Zygote = "e88e6eb3-aa80-5325-afca-941959d7151f"
 
 [extensions]
-OptimizationEnzymeExt = "Enzyme"
-OptimizationFiniteDiffExt = "FiniteDiff"
-OptimizationForwardDiffExt = "ForwardDiff"
+OptimizationDIExt = "DifferentiationInterface"
 OptimizationMTKExt = "ModelingToolkit"
-OptimizationReverseDiffExt = "ReverseDiff"
-OptimizationSparseDiffExt = ["SparseDiffTools", "Symbolics", "ReverseDiff"]
-OptimizationTrackerExt = "Tracker"
-OptimizationZygoteExt = "Zygote"
 
 [compat]
 ADTypes = "0.2.5"
 ArrayInterface = "7.6"
 DocStringExtensions = "0.9"
-Enzyme = "0.11.11"
-FiniteDiff = "2.12"
-ForwardDiff = "0.10.26"
 LinearAlgebra = "1.9, 1.10"
-ModelingToolkit = "8.74, 9"
+ModelingToolkit = "9"
```

Comment:
would this fit into DI as well?

Reply:
Yeah a version of this might make sense there. In DI it'd probably just be Symbolics.jl and not mtk like here, since this uses the https://github.com/SciML/OptimizationBase.jl/pull/31/files#diff-9535d17a8cb1631f5b7f0c2ef812c0dd8efd722128cfddf059baab25976c7ed3L66 directly so the relevant code to be moved is this https://github.com/SciML/ModelingToolkit.jl/blob/9bfe88015ba2101218fa68c40323b55eaf04f2a5/src/systems/optimization/optimizationsystem.jl#L129-L159, so it can be adapted to the sparsity API as well